src/content/getting-started/ethereum-glossary.md

```diff
@@ -200,8 +202,58 @@ Usually, the person that is encrypting the data will possess the key that locks
 
 * A piece of code (or program) that is stored on the blockchain network. Conditions of the contract are predefined by the users, if all conditions are met, certain actions are executed by the contract (program).
 
+* These can be executed automatically by the same distributed blockchain that it is stored on.
+
+* These contracts are designed to be tamper-proof permanent (which includes the code used in their creation). To edit a contract, one will have to create and deploy an entirely new contract.
+
 ### Blockchain
 
 * A decentralized publicly owned ledger.
 
-All feedback, rewrites, clarification, typo-fixing, and requests for additions are more than welcome. ?
+* Consists of a series of Blocks.
+
+### Block
+
+* A data structure that contains aggregated transaction data.
+
+* Is cryptographically linked together to form the blockchain (literally, a chain of blocks).
+
+* The first block in a blockchain is called the Genesis Block.
+
+### Hashing
+
+* Calculating a function that takes an arbitrary amount of input data and deterministically produces a fixed-length output known as the data's "hash".
+
+* This can be used to verify that data has not be altered as any part of the input data is changed, the hash will change too.
+
+### Mining
+
+* The process by which transactions are verified and added to a Proof-of-Work blockchain.
+
+* Solving cryptographic problems using high-powered, specialized computing hardware is the actual process that is being completed.
+
+### Proof-of-Work
+
+* A system of using to computational power to apply a method of security to the consensus of a blockchain (determining the order of blocks in a blockchain).
+
+* Blocks of transactions must be hashed with an additional parameter which takes many tries to create a valid block hash.
+
+* The successfully hashed block is considered a correct and valid proof of computational work.
+
+* An alternative to this system is called Proof-Of-Work.
```

Comment:
Proof of stake? Might also be nice to put an entry in here for that.

Reply:
Aye. Thanks.